src/core_modules/capture-core/components/DataEntries/SingleEventRegistrationEntry/DataEntryWrapper/DataEntry/epics/newEventDataEntry.epics.ts

```diff
@@ -45,10 +45,18 @@ type SelectionsCompletenessPayload = {
     triggeringActionType?: string;
 };
 
-export const resetDataEntryForNewEventEpic = (action$: EpicAction<any>) =>
+export const resetDataEntryForNewEventEpic = (action$: EpicAction<any>, store: ReduxStore) =>
     action$.pipe(
         ofType(newEventDataEntryBatchActionTypes.SAVE_NEW_EVENT_ADD_ANOTHER_BATCH),
-        map(() => (batchActions(getOpenDataEntryActions()))),
+        map(() => {
+            const state = store.value;
+            const orgUnitId = state.currentSelections.orgUnitId;
+            const orgUnits = state.organisationUnits;
+            const orgUnit = orgUnitId && orgUnits
+                ? orgUnits[orgUnitId]
+                : undefined;
+            return batchActions(getOpenDataEntryActions(orgUnit));
```

Comment:
To retain the categories, `programCategory` and `selectedCategories` also need to be passed to getOpenDataEntryActions. You can achieve this by following the existing approach used in `openSingleEventDataEntry`.

https://github.com/user-attachments/assets/2a205c75-0425-406e-abd5-0fb1e45a5020